java/ql/src/semmle/code/java/frameworks/play/PlayHTTPRequestHeader.qll

```diff
@@ -0,0 +1,25 @@
+import java
+
+/**
+ * Play MVC Framework HTTP Request Header Class
+ */
+class PlayMVCHTTPRequestHeader extends RefType {
+  PlayMVCHTTPRequestHeader() { this.hasQualifiedName("play.mvc", "Http$RequestHeader") }
+}
+
+/**
+ * Play Framework HTTPRequestHeader Methods - `headers`, `getQueryString`, `getHeader`
+ *
+ * Documentation: https://www.playframework.com/documentation/2.6.0/api/java/play/mvc/Http.RequestHeader.html
+ */
+class PlayMVCHTTPRequestHeaderMethods extends Method {
+  PlayMVCHTTPRequestHeaderMethods() { this.getDeclaringType() instanceof PlayMVCHTTPRequestHeader }
+
+  /**
+   * Gets all references to play.mvc.HTTP.RequestHeader `getQueryString` method
+   */
+  MethodAccess getQueryString() {
+    this.hasName("getQueryString") and result = this.getAReference()
```

Comment:
If you just want to model this single method for now, push the `this.hasName("getQueryString")` into the `PlayMVCHTTPRequestHeaderMethods()` predicate above and rename the class to `PlayMVCHTTPRequestHeaderQueryString` or similar

Reply:
@smowton not really the idea, was `HTTPRequestHeader` has different methods `headers`, `getQueryString`, `getHeader`.  `PlayMVCHTTPRequestHeaderMethods` allows to access these mentioned and since only `getQueryString` was being actively model'd i just added it as as member predicate.